src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx

```diff
@@ -1,37 +1,127 @@
+import { SelectListItemFragmentContainer } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListItem"
+import { SelectListsForArtworkHeaderFragmentContainer } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkHeader"
 import { SelectListsForArtworkFooter } from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkFooter"
-import { times } from "lodash"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import { createFragmentContainer, graphql } from "react-relay"
+import { SelectListsForArtworkModal_me$data } from "__generated__/SelectListsForArtworkModal_me.graphql"
+import { SelectListsForArtworkModal_artwork$data } from "__generated__/SelectListsForArtworkModal_artwork.graphql"
+import { SelectListsForArtworkModalQuery } from "__generated__/SelectListsForArtworkModalQuery.graphql"
+import { extractNodes } from "Utils/extractNodes"
+import {
+  SelectListsForArtworkHeaderPlaceholder,
+  SelectListsPlaceholder,
+} from "Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkPlaceholders"
+import { getSelectedCollectionIds } from "Apps/CollectorProfile/Routes/Saves2/Utils/getSelectedCollectionIds"
+
+interface SelectListsForArtworkModalQueryRenderProps {
+  artworkID: string
+  onClose: () => void
+}
 
 export interface SelectListsForArtworkModalProps {
+  me: SelectListsForArtworkModal_me$data | null
+  artwork: SelectListsForArtworkModal_artwork$data | null
   onClose: () => void
 }
 
 export const SelectListsForArtworkModal: React.FC<SelectListsForArtworkModalProps> = ({
+  me,
+  artwork,
   onClose,
 }) => {
-  const [selectedIds, setSelectedIds] = useState<string[]>([])
+  const [addToCollectionIDs, setAddToCollectionIDs] = useState<string[]>([])
+  const [removeFromCollectionIDs, setRemoveFromCollectionIDs] = useState<
+    string[]
+  >([])
+  const savedCollection = me?.defaultSaves
+  const nodes = extractNodes(me?.collectionsConnection)
+  const collections = savedCollection ? [savedCollection, ...nodes] : nodes
+  const selectedCollectionIds = getSelectedCollectionIds({
+    collections,
+    addToCollectionIDs,
+    removeFromCollectionIDs,
+  })
+  const hasChanges =
+    addToCollectionIDs.length !== 0 || removeFromCollectionIDs.length !== 0
+
+  const addOrRemoveCollectionIdFromIds = (
+    ids: string[],
+    collectionId: string
+  ) => {
+    if (ids.includes(collectionId)) {
+      return ids.filter(id => id !== collectionId)
+    }
+
+    return [...ids, collectionId]
+  }
 
-  const handleItemSelected = (selectedId: string) => {
-    if (selectedIds.includes(selectedId)) {
-      const updatedSelectedIds = selectedIds.filter(id => id !== selectedId)
+  const handleItemPress = (item: typeof collections[0]) => {
+    if (item.isSavedArtwork) {
+      const updatedIds = addOrRemoveCollectionIdFromIds(
```

Comment:
**suggestion / non-blocking**: in my opinion, `addOrRemoveCollectionIdFromIds` is a bit redundant, since we use it in two places - in one to add collection, in another one - to remove from collection. Will it not be easier to do something like this?

```javascript
const updatedRemoveFromCollectionIds = removeFromCollectionIDs.filter(id => id !== item.internalID)
setRemoveFromCollectionIDs(updatedRemoveCollectionIds)
```

and

```javascript
const updatedAddCollectionIds = [...addToCollectionIDs, item.internalID]
setAddToCollectionIDs(updatedRemoveCollectionIds)
```

Reply:
there is a slightly different logic here. If `isSavedArtwork` is `true`, then we have to work with `removeFromCollectionIDs` list (the user can unselect the previously selected default list and re-select it). If `isSavedArtwork` is `fakse`, then we have to work with `addToCollectionIDs ` list. 

And in order not to duplicate the logic of select/unselect elements for these two list, `addOrRemoveCollectionIdFromIds` was added

Reply:
If we want to remove `addOrRemoveCollectionIdFromIds`, we will end up with this code

```javascript
const handleItemPress = (item: typeof collections[0]) => {
  if (item.isSavedArtwork) {
    if (removeFromCollectionIDs.includes(item.internalID)) {
      setRemoveFromCollectionIDs(removeFromCollectionIDs.filter(id => id !== item.internalID))
      return
    }

    setRemoveFromCollectionIDs([...removeFromCollectionIDs, item.internalID])
    return
  }

  if (addToCollectionIDs.includes(item.internalID)) {
    setAddToCollectionIDs(addToCollectionIDs.filter(id => id !== item.internalID))
    return
  }

  setAddToCollectionIDs([...addToCollectionIDs, item.internalID])
}
```

Reply:
understood! Does it makes sense to add a few lines explaining the code then?

Reply:
Yes, that's a good suggestion. I'll add it later